xvec/accessor.py

```diff
@@ -494,3 +499,79 @@ def set_crs(
             _obj = _obj.set_xindex(key, GeometryIndex, crs=crs)
 
         return _obj
+
+    def set_geom_indexes(
+        self,
+        coord_names: str | Sequence[Hashable],
+        crs: Any = None,
+        allow_override: bool = False,
+        **kwargs,
+    ):
+        """Set a new  :class:`~xvec.GeometryIndex` for one or more existing
```

Comment:
Given that the Xarray allows multi-coordinate indexes, the description is slightly misleading. It should clarify that one index is set per coordinate.

Also probably worth adding that only 1d coordinates are supported.